src/schema/term.rs

```diff
@@ -76,10 +86,22 @@ impl Term {
     /// The first byte is 2, and the three following bytes are the utf-8 
     /// representation of "abc".
     pub fn from_field_text(field: Field, text: &str) -> Term {
-        let mut buffer = allocate_vec(4 + text.len());
-        BigEndian::write_u32(&mut buffer[0..4], field.0);
-        buffer[4..].clone_from_slice(text.as_bytes());
-        Term(buffer)
+        let buffer = Vec::with_capacity(4 + text.len());
+        let mut term = Term(buffer);
+        term.set_field(field);
+        term.set_text(text);
+        term
+    }
+
+    /// Creates a new Term with an empty buffer, 
+    /// but with a given capacity.
+    ///
+    /// It is declared unsafe, as the term content
+    /// is not initialized, and a call to `.field()`
+    /// would panic.
+    #[doc(hidden)]
+    pub unsafe fn with_capacity(num_bytes: usize) -> Term {
```

Comment:
Oh this is the first time I hear about `pub(crate)` ! I have been wishing for something like for so long. Awesome.

Wouldn't `Term::uninitialized` suggest that the buffer is unitialized and with a len of  `num_bytes`?